packages/graph/hash_graph/lib/graph/src/datastore/mod.rs

```diff
@@ -154,7 +157,7 @@ trait Datastore {
     ///
     /// - [`DatastoreError`], if the account referred to by `created_by` does not exist.
     async fn create_data_type(
-        &self,
+        &mut self,
```

Comment:
I assume this change is made such that other Datastore implementations can still get to modify the implementor. This wasn't added because of something needed by sqlx, correct?

Reply:
Yup, `sqlx` doesn't require the database to be mutable, but for our interface, we should enforce it.